code/game/mecha/mecha.dm

```diff
@@ -565,6 +565,8 @@
 
 	if(prob(deflect_chance * deflection) && (Proj.damage_type == BRUTE || Proj.damage_type == BURN))

```

Comment:
The brue and burn check should be done separately and before the deflection stuff, that way only projectiles that do brute and burn damage will go through.